CMakeLists.txt

```diff
@@ -134,7 +134,7 @@ if(WITH_GZFILEOP)
     add_definitions(-DWITH_GZFILEOP)
 endif()
 
-if("${CMAKE_C_COMPILER}" MATCHES "icc" OR "${CMAKE_C_COMPILER}" MATCHES "icpc" OR "${CMAKE_C_COMPILER}" MATCHES "icl")
+if(("${CMAKE_C_COMPILER}" MATCHES "icc" AND NOT "${CMAKE_C_COMPILER}" MATCHES "mpicc") OR "${CMAKE_C_COMPILER}" MATCHES "icpc" OR "${CMAKE_C_COMPILER}" MATCHES "icl")
```

Comment:
I would rather see mpicc having own section before section for icc... That way this test doesn't become hard to read.

That new section would list correct flags that mpicc supports.

Reply:
mpicc is typically a wrapper around either gcc or clang or intel or ibm compilers.  Would need to be able to somehow determine the underlying compiler (which is already done by CMake:
```
s1041980:zlib-ng-2.0.2(master)> CC=mpicc cmake -Bbuild -DCMAKE_INSTALL_PREFIX=${INSTALL_PATH} -DZLIB_COMPAT=YES .
-- Using CMake version 3.19.8
-- ZLIB_HEADER_VERSION: 1.2.11
-- ZLIBNG_HEADER_VERSION: 2.0.2
-- The C compiler identification is Clang 8.0.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/local/bin/mpicc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
```

So maybe the test needs to be done on whatever symbol is set for `C compiler identification`?   I will look closer to see how to make this cleaner.

Reply:
As far as I know, MPI has own package for CMake, so overriding `CC` and `CXX` is discouraged.

Reply:
In all of the parallel MPI builds that we do (Trilinos, SEACAS, HDF5, NetCDF), we set the compiler to `mpicc` and `mpicxx`.  CMake does recognize that this is a parallel compiler and handles it from there...